### ISSUE TITLE
Use two datePickers with range selection mode

### ISSUE THREAD

**Issue by zdrawku** [IgniteUI/igniteui-angular-marketing-dashboard#13]
We should use only two datePickers instead of four. @HristoP96 replace them with only two datePickrs that are using the range mode. The end-user should be able to select two ranges and compare them.

Also, I don't like the strict date selection that is currently used. For example, if a year is selected, the ranges between dates should be exactly 365 days and the endDay of the first range needs to be the same as the startDay of the second range, this limits the behavior and is not intuitive, it took me minutes and a lot of tweaks in order to understand what should I do, we should definitely change that and make it more user friendly.

In conclusion:

Use only two datePickers with range selection
do not limit the user to select strict ranges, the user should be able to select custom ranges despite of the selected type (1 year, 3 months, 1 month or 1 week)

**Comment by HristoP96** [comment]
Due to performance issues with the calendars, this fix for this issue must be delayed until a solution for a better performance is introduced

**Comment by StefanIvanov** [comment]
@HristoP96 could you elaborate on the specifics of the performance issue that you are hitting?

**Comment by HristoP96** [comment]
> @HristoP96 could you elaborate on the specifics of the performance issue that you are hitting?

@StefanIvanov When a big range is selected(e.g. 3 years) from the Start range calendar and compared, the new states of all the components are rendered really slow. This behavior is not reproducible if a big date range is compared, through the datepickers.

**Comment by StefanIvanov** [comment]
To me this sounds like a potential performance issue in the range selection date picker that we will need to investigate anyways. Could you please investigate how the two behaviors are different to affect performance so prominently?
Will a workaround like the following solve the performance issue: components are rendered only once the "Compare" button has been clicked.

**Comment by HristoP96** [comment]
After investigating the calendar's range selection  behavior, it seems that problem is with the change detection. I observed how some functions bound to the bullet graphs inputs are invoked hundreds of times after a big range selection is made. It seems the selection of each date in the range selection iteration triggers change detection.

I will try to workaround this with an OnPush change detection strategy.

**Comment by StefanIvanov** [comment]
Please do so :)
